### PR TITLE
Add maxmemory value to the Metric ingestion from Redis Info.

### PR DESCRIPTION
**Pull request by katzchang**
#### Description of the changes

Add maxmemory value to the Metric ingestion from Redis Info. This allows users to calculate the saturation rate of Redis memory.

~There's a change in indentation at a point unrelated to the purpose, because when I gofmt it, I had to make some changes to the indentation.~

### Author

- [x] the PR should focus on a single subject. Change only relevant files to the problem you’re working on
- [x] Clean and format the code
- [x] add unit tests for your changes and make sure all unit tests are passing
- [x] add a risk label after carefully considering the "blast radius" of your changes
- [x] describe the _intent_ of your changes in the description. don't just rewrite your code in prose
- [x] assign at least one reviewer
- [ ] address the feedback 

### Reviewer

- [ ] review code for readability
- [ ] verify that high risk behavior changes are well tested
- [ ] check license for any new external dependency
- [ ] ask questions about anything that isn't clear and obvious
- [ ] approve the PR when you consider it's good to merge

### CODE REVIEW

File: src/metrics.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ var metricsDefinition = map[string][]interface{}{
 	"system.usedMemoryPeakBytes":    {"used_memory_peak", metric.GAUGE},
 	"system.usedMemoryLuaBytes":     {"used_memory_lua", metric.GAUGE},
 	"system.totalSystemMemoryBytes": {"total_system_memory", metric.GAUGE},
+	"system.maxmemory":              {"maxmemory", metric.GAUGE},
```
Comment (gsanchezgavier) [inline src/metrics.go:31]: Is this metrics unit Bytes? The unit should be added at the end as per the other memory metrics

Comment (katzchang) [reply src/metrics.go]: > maxmemory: The value of the maxmemory configuration directive
> https://redis.io/commands/info

It's not explicitly stated in the official reference - I'm guessing it's bytes, just like total_system_memory, etc.

Tried:
```
config get maxmemory
maxmemory
0
...
config set maxmemory 128M
+OK

config get maxmemory '
maxmemory
128000000

info
...
used_memory_lua_human:37.00K
maxmemory:128000000
maxmemory_human:122.07M
maxmemory_policy:noeviction
...
```

**Comment by katzchang** [comment]
Hi @davidbrota, 
I'm waiting for your review, when will it be? Thank you!

File: src/metrics.go (hunk 1 above)
Comment (davidbrota) [reply src/metrics.go]: I would also put the Bytes suffix like the other metrics

**Comment by ardias** [comment]
Hi @katzchang 
I'm filling in for @davidbrota since he's on vacations I believe. 
Looks good but I would ask a small change in the "output" metric name. All the other metrics use camelCase for the name, so I think it look better if we did the same thing here:

maxmemory -> maxMemoryBytes

What do you think?

**Comment by katzchang** [comment]
Thanks @davidbrota, 

> LGTM, could you please add the suffix Bytes to the new metric and update the changelog too @katzchang?

Got it, I'll try.

**Comment by katzchang** [comment]
@ardias,

> maxmemory -> maxMemoryBytes

I thought about this too, but since the original Redis name of the item is `maxmemory` instead of `max_memory`, I chose to follow it.

It's not a strong opinion, so I'll stick with it.

Thanks.

**Review by ardias — APPROVED** [review]

**Pull request merged**